### PR TITLE
Add remediations page

### PR DESCRIPTION
**Pull request by Monnte**
**OAMG-9902**

Added page for remediations.
Table with available remediation with some metadata and link to them, introduced in this PR.

Additional filtering for leapp remediations playbooks will be needed (another task).

- [x] PR has a short description
- [x] Tests for the changes have been added

### CODE REVIEW

File: src/Helpers/constants.ts [hunk 1]
```diff
@@ -26,5 +27,6 @@ export const RECOMMENDATIONS_SYSTEM_DETAILS = '/systems_detail/';
 export const REMEDIATIONS_API_ROOT = '/api/remediations/v1';
 export const REMEDIATIONS_RESOLUTIONS_ROOT = '/resolutions';
 export const REMEDIATIONS_REMEDIATIONS_ROOT = '/remediations';
+export const REMMEDIATIONS_DETAIL_ROOT = '/insights/remediations';
```
Comment (andywaltlova) [inline src/Helpers/constants.ts:30]:
```suggestion
export const REMEDIATIONS_DETAIL_ROOT = '/insights/remediations';
```

File: src/Components/RemediationsPage/__tests__/RemediationsPage.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  usePermissions: jest.fn(),
+}));
+jest.mock('axios');
+
+describe('RemediationsPage', () => {
+  beforeEach(() => {
+    usePermissions.mockImplementation(() => ({ hasAccess: true, isLoading: false }));
+  });
+
+  it('should render table with recommendations', async () => {
```
Comment (andywaltlova) [inline src/Components/RemediationsPage/__tests__/RemediationsPage.test.js:20]: Shouldn't this whole file use the remediation word isntead of recommendation? I assume it was just copied from recommendations :sweat_smile:

File: src/Components/RemediationsPage/RemediationsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    description: 'Name',
+    updated_at: 'Last modified',
+    status: 'Issues resolved',
+    system_count: 'Total systems',
+  };
+
+  const [showSkeleton, setShowSkeleton] = useState(false);
+  const [activeSortIndex, setActiveSortIndex] = React.useState<number | null>(null);
+  const [activeSortDirection, setActiveSortDirection] = React.useState<'asc' | 'desc' | null>(null);
+  const [remmediations, setRemediations] = React.useState<Remediation[]>([]);
```
Comment (andywaltlova) [inline src/Components/RemediationsPage/RemediationsTable.tsx:28]:
```suggestion
  const [remediations, setRemediations] = React.useState<Remediation[]>([]);
```

File: src/Components/RemediationsPage/RemediationsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            <Th sort={getSortParams(1)}>{columnNames.updated_at}</Th>
+            <Th>{columnNames.status}</Th>
+            <Th sort={getSortParams(2)}>{columnNames.system_count}</Th>
+          </Tr>
+        </Thead>
+        {showSkeleton ? (
+          loadingSkeletons(perPage, Object.keys(columnNames).length)
+        ) : (
+          <Tbody>
+            {remmediations.length === 0 ? (
```
Comment (andywaltlova) [inline src/Components/RemediationsPage/RemediationsTable.tsx:106]:
```suggestion
            {remediations.length === 0 ? (
```

File: src/Components/RemediationsPage/RemediationsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                <Td colSpan={Object.keys(columnNames).length}>
+                  <Bullseye>
+                    <EmptyState variant={EmptyStateVariant.sm}>
+                      <EmptyStateHeader icon={<EmptyStateIcon icon={SearchIcon} />} titleText="No remediations found" headingLevel="h2" />
+                    </EmptyState>
+                  </Bullseye>
+                </Td>
+              </Tr>
+            ) : (
+              remmediations.map((rec, rowIndex) => (
```
Comment (andywaltlova) [inline src/Components/RemediationsPage/RemediationsTable.tsx:117]:
```suggestion
              remediations.map((rec, rowIndex) => (
```

File: src/Components/RemediationsPage/RemediationsTable.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+                <Tr key={rowIndex}>
+                  <Td dataLabel={columnNames.description} width={60}>
+                    <a key={rec.id} href={`${REMMEDIATIONS_DETAIL_ROOT}/${rec.id}`} target="_blank" rel="noreferrer">
```
Comment (andywaltlova) [inline src/Components/RemediationsPage/RemediationsTable.tsx:120]:
```suggestion
                    <a key={rec.id} href={`${REMEDIATIONS_DETAIL_ROOT}/${rec.id}`} target="_blank" rel="noreferrer">
```

File: src/Components/RemediationsPage/RemediationsTable.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        ) : (
+          <Tbody>
+            {remmediations.length === 0 ? (
+              <Tr>
+                <Td colSpan={Object.keys(columnNames).length}>
+                  <Bullseye>
+                    <EmptyState variant={EmptyStateVariant.sm}>
+                      <EmptyStateHeader icon={<EmptyStateIcon icon={SearchIcon} />} titleText="No remediations found" headingLevel="h2" />
+                    </EmptyState>
+                  </Bullseye>
```
Comment (andywaltlova) [inline src/Components/RemediationsPage/RemediationsTable.tsx:113]: Little bit prefer what actual remediations have, it's more informative

image

**Review by andywaltlova — CHANGES_REQUESTED** [review]
Mostly just typo commentc, otherwise LGTM, created two other issues in repo based on the testing I did, feel free to comment on them.

File: src/Components/RemediationsPage/RemediationsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+      system_count: rem.system_count,
+      resolved_count: rem.resolved_count,
+    }));
+
+    setRemediations(remediations);
+    const count = response.meta.count;
+    setTotal(count);
+
+    setShowSkeleton(false);
+    return count;
```
[the remark this answers is not in the document]
Comment (Monnte) [reply src/Components/RemediationsPage/RemediationsTable.tsx:69]: Yep, we don't need that return value.

**Review by andywaltlova — APPROVED** [review]

**Pull request merged**